### ISSUE TITLE
[FIXED] `error bundler: failed to load command: pod`, `deprecate to_default_s: :to_s, deprecator: ActiveSupport.deprecator`

### ISSUE THREAD

**Issue by Angelk90** [facebook/react-native#39832]
### Description

I tried creating a new project in react native:

` npx react-native@latest init myApp
`

When creating the project I get this:

```
Installing CocoaPods dependencies (this may take a few minutes)
error bundler: failed to load command: pod (/Users/nameUser/Desktop/proj/myApp/vendor/bundle/ruby/3.2.0/bin/pod)
/Users/nameUser/Desktop/proj/myApp/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:108:in `<class:Array>': undefined method `deprecator' for ActiveSupport:Module (NoMethodError)

  deprecate to_default_s: :to_s, deprecator: ActiveSupport.deprecator
                                                          ^^^^^^^^^^^
Did you mean?  deprecate_constant
    from /Users/nameUser/Desktop/proj/myApp/vendor/bundle/ruby/3.2.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:8:in `<top (required)>'
    from <internal:/Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
    from <internal:/Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
    from /Users/nameUser/Desktop/proj/myApp/vendor/bundle/ruby/3.2.0/gems/cocoapods-1.13.0/lib/cocoapods.rb:9:in `<top (required)>'
    from <internal:/Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
    from <internal:/Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:38:in `require'
    from /Users/nameUser/Desktop/proj/myApp/vendor/bundle/ruby/3.2.0/gems/cocoapods-1.13.0/bin/pod:36:in `<top (required)>'
    from /Users/nameUser/Desktop/proj/myApp/vendor/bundle/ruby/3.2.0/bin/pod:25:in `load'
    from /Users/nameUser/Desktop/proj/myApp/vendor/bundle/ruby/3.2.0/bin/pod:25:in `<top (required)>'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/cli/exec.rb:58:in `load'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/cli/exec.rb:58:in `kernel_load'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/cli/exec.rb:23:in `run'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/cli.rb:492:in `exec'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/cli.rb:34:in `dispatch'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/cli.rb:28:in `start'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.20/exe/bundle:37:in `block in <top (required)>'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/bundler/friendly_errors.rb:117:in `with_friendly_errors'
    from /Users/nameUser/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/bundler-2.4.20/exe/bundle:29:in `<top (required)>'
    from /Users/nameUser/.rbenv/versions/3.2.2/bin/bundle:25:in `load'
    from /Users/nameUser/.rbenv/versions/3.2.2/bin/bundle:25:in `<main>'
✖ Installing CocoaPods dependencies (this may take a few minutes)
error Looks like your iOS environment is not properly set. Please go to https://reactnative.dev/docs/environment-setup?os=macos&platform=android and follow the React Native CLI QuickStart guide for macOS and iOS.
```

```json
  "dependencies": {
    "react": "18.2.0",
    "react-native": "0.72.5"
  }
```

### React Native Version

0.72.5

### Output of `npx react-native info`

```
System:
  OS: macOS 14.0
  CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
  Memory: 2.95 GB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 18.17.1
    path: ~/.nvm/versions/node/v18.17.1/bin/node
  Yarn:
    version: 1.22.4
    path: /usr/local/bin/yarn
  npm:
    version: 9.6.7
    path: ~/.nvm/versions/node/v18.17.1/bin/npm
  Watchman:
    version: 2023.09.04.00
    path: /usr/local/bin/watchman
Managers:
  CocoaPods:
    version: 1.13.0
    path: /Users/nameUser/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 23.0
      - iOS 17.0
      - macOS 14.0
      - tvOS 17.0
      - watchOS 10.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10811636
  Xcode:
    version: 15.0/15A240d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 20.0.1
    path: /usr/bin/javac
  Ruby:
    version: 3.2.2
    path: /Users/nameUser/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.5
    wanted: 0.72.5
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: Not found
  newArchEnabled: Not found
```

### Steps to reproduce

I tried creating a new project in react native:

` npx react-native@latest init myApp
`

**Comment by stephen-pearce** [comment]
@Angelk90 That looks to be a solution to the problem if I understand it right. Hopefully this gets into a stable release soon

**Comment by radetsky** [comment]
I have the same thing. The solution suggested @stephen-pearce fix it. Thank you.

**Comment by Will-lopesz** [comment]
I'm having the same problem, when running init it gives the error, but it still creates the project, I spent the day thinking it was a problem with my machine, but no matter what I do nothing seems to solve it. I'm going to try the solution suggested above, I'm just afraid of having more problems with this in the future.

**Comment by cipolleschi** [comment]
Yeah, so... Unfortunately, Active Support released version 7.1.0 this morning, which is not compatible with Cocoapods.
I already pushed a fix on main: https://github.com/facebook/react-native/commit/ce39931bc2b02f13cbc5751ba4d4a6dbc07bc91a 
The release crew is aware of that and they are working to release all the supported versions with the fix:
- 0.70
- 0.71
- 0.72
- 0.73.0-RC.2

Meanwhile, you can:
- apply the changes shown in the commit above to your local Gemfile
- run `bundle install` again to pull the right version of ActiveSupport (this is needed only once)
- run `bundle exec pod install` to reinstall the pods.

I'm sorry for the inconvenience and thank you for your patience!

**Comment by mvalbuquerque** [comment]
Help! 

 Downloading template
✔ Copying template
✔ Processing template
✔ Installing Ruby Gems
✖ Installing CocoaPods dependencies (this may take a few minutes)
error bundler: failed to load command: pod (/Users/username/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/bin/pod)
/Users/username/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:108:in `<class:Array>': undefined method `deprecator' for ActiveSupport:Module (NoMethodError)

  deprecate to_default_s: :to_s, deprecator: ActiveSupport.deprecator
                                                          ^^^^^^^^^^^
Did you mean?  deprecate_constant
        from /Users/mvalbuquerque/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/gems/activesupport-7.1.0/lib/active_support/core_ext/array/conversions.rb:8:in `<top (required)>'
        from /Users/username/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods.rb:9:in `require'
        from /Users/username/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/gems/cocoapods-1.13.0/lib/cocoapods.rb:9:in `<top (required)>'
        from /Users/username/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/gems/cocoapods-1.13.0/bin/pod:36:in `require'
        from /Users/username/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/gems/cocoapods-1.13.0/bin/pod:36:in `<top (required)>'
        from /Users/username/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/bin/pod:25:in `load'
        from /Users/username/Desktop/Apps/AppFormulario/vendor/bundle/ruby/3.1.0/bin/pod:25:in `<top (required)>'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `load'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:58:in `kernel_load'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/cli/exec.rb:23:in `run'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/cli.rb:484:in `exec'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/cli.rb:31:in `dispatch'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/cli.rb:25:in `start'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/libexec/bundle:48:in `block in <top (required)>'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/3.1.0/bundler/friendly_errors.rb:103:in `with_friendly_errors'
        from /Users/username/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/bundler-2.3.7/libexec/bundle:36:in `<top (required)>'
        from /Users/username/.rbenv/versions/3.1.2/bin/bundle:25:in `load'
        from /Users/username/.rbenv/versions/3.1.2/bin/bundle:25:in `<main>'

**Comment by cipolleschi** [comment]
@mvalbuquerque if you need to be unblocked:

* apply the changes shown in thece39931bc2b02f13cbc5751ba4d4a6dbc07bc91a to your local Gemfile. 
* Delege the Gemfile.lock
* run bundle install again to pull the right version of ActiveSupport (this is needed only once)
* run bundle exec pod install to reinstall the pods.

**Comment by kyadalu1** [comment]
> * [39931bc](https://github.com/facebook/react-native/commit/39931bc2b02f13cbc5751ba4d4a6dbc07bc91a)

Shows 404

**Comment by muhammedyasirmt** [comment]
@kyadalu1 

- add `gem 'activesupport', '>= 6.1.7.3', '< 7.1.0'` to Gemfile.
- delete Gemfile.lock.
- run `cd ios && bundle install && cd ..` in ios folder.
- run `npx pod-install`

**Comment by cipolleschi** [comment]
Thanks @muhammedyasirmt, that's the right fix.
Sorry @kyadalu1 for the 404.

**Comment by DominicTobias** [comment]
I get this error just running `npx react-native@latest init AwesomeProject` tried ruby 2.7.4 and 3.2.2 :|

**Comment by dpitts01** [comment]
@stephen-pearce Thanks a ton, works for us, too! Really appreciate this quick response.

**Comment by mvalbuquerque** [comment]
Hello! [cipolleschi](https://github.com/cipolleschi) Thank you for your answer, the solution works for me, but it doesn't work when I run the command npx react-native@latest init AwesomeProject, I need to generate many projects in my classes, could you let me know when the fix comes out?

**Comment by mvalbuquerque** [comment]
Hello! This worked for me but not when creating the project :-(



[alternative after generating the project](https://stackoverflow.com/questions/77236339/after-updating-cocoapods-to-1-13-0-it-throws-error)

**Comment by cipolleschi** [comment]
>  I need to generate many projects in my classes

What do you mean? If you have to generate many projects within the same React Native app, all the projects must have the updated `ActiveSupport` line.

Anyway, the Release crew is aiming to have 0.71 and 0.72 by the end of this week, if everything runs smoothly

**Comment by cipolleschi** [comment]
Opening this to keep it as main issue for tracking the ActiveSupport problem.

**Comment by kelset** [comment]
just a quick update on this folks: patch releases for 71 and 72 are in the works as we speak. 0.70 will follow soon after.

**Comment by vksgautam1986** [comment]
Command --npx react-native init awesomeProject
Got this issue
error @react-native/normalize-colors@0.74.0: The engine "node" is incompatible with this module. Expected version ">=18". Got "16.20.2"
error Found incompatible module.

**Comment by huntie** [comment]
## Resolved — please upgrade to `react-native@0.72.6`

We've shipped a fix for this bug as part of the latest React Native 0.72 patch release: https://github.com/facebook/react-native/releases/tag/v0.72.6.

Please upgrade to this version to resume successful iOS builds!

**Comment by Prabhat1933** [comment]
While Installing Bundle in Ios Directory Give following error
Fetching gem metadata from https://rubygems.org/.......
Installing ffi 1.16.3 with native extensions
--- ERROR REPORT TEMPLATE -------------------------------------------------------

```
Gem::FilePermissionError: You don't have write permissions for the /Users/user/Documents/Awesome/vendor/bundle/ruby/3.2.0 directory.
  /Users/user/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/installer.rb:713:in `verify_gem_home'
  /Users/user/.rbenv/versions/3.2.2/lib/ruby/site_ruby/3.2.0/rubygems/installer.rb:903:in `pre_install_checks'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/rubygems_gem_installer.rb:64:in `pre_install_checks'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/rubygems_gem_installer.rb:12:in `install'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/source/rubygems.rb:203:in `install'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/installer/gem_installer.rb:54:in `install'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/installer/gem_installer.rb:16:in `install_from_spec'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/installer/parallel_installer.rb:130:in `do_install'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/installer/parallel_installer.rb:121:in `block in worker_pool'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/worker.rb:62:in `apply_func'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/worker.rb:57:in `block in process_queue'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/worker.rb:54:in `loop'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/worker.rb:54:in `process_queue'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/worker.rb:90:in `block (2 levels) in create_threads'
Errno::EACCES: Permission denied @ rb_sysopen - /Users/user/Documents/Awesome/vendor/bundle/ruby/3.2.0/bundler.lock
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/process_lock.rb:9:in `initialize'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/process_lock.rb:9:in `open'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/process_lock.rb:9:in `lock'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/installer.rb:71:in `run'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/installer.rb:23:in `install'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/cli/install.rb:62:in `run'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/cli.rb:261:in `block in install'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/settings.rb:142:in `temporary'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/cli.rb:260:in `install'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/cli.rb:34:in `dispatch'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/cli.rb:28:in `start'
  /Users/user/.gem/gems/bundler-2.4.21/exe/bundle:37:in `block in <top (required)>'
  /Users/user/.gem/gems/bundler-2.4.21/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
  /Users/user/.gem/gems/bundler-2.4.21/exe/bundle:29:in `<top (required)>'
  /Users/user/.rbenv/versions/3.2.2/bin/bundle:25:in `load'
  /Users/user/.rbenv/versions/3.2.2/bin/bundle:25:in `<main>'

```

## Environment

```
Bundler       2.4.21
  Platforms   ruby, x86_64-darwin-22
Ruby          3.2.2p53 (2023-03-30 revision e51014f9c05aa65cbf203442d37fef7c12390015) [x86_64-darwin-22]
  Full Path   /Users/user/.rbenv/versions/3.2.2/bin/ruby
  Config Dir  /Users/user/.rbenv/versions/3.2.2/etc
RubyGems      3.4.21
  Gem Home    /Users/user/Documents/Awesome/vendor/bundle/ruby/3.2.0
  Gem Path    /Users/user/Documents/Awesome/vendor/bundle/ruby/3.2.0
  User Home   /Users/user
  User Path   /Users/user/.gem/ruby/3.2.0
  Bin Dir     /Users/user/Documents/Awesome/vendor/bundle/ruby/3.2.0/bin
OpenSSL       
  Compiled    OpenSSL 3.1.3 19 Sep 2023
  Loaded      OpenSSL 3.1.3 19 Sep 2023
  Cert File   /Users/user/.rbenv/versions/3.2.2/openssl/ssl/cert.pem
  Cert Dir    /Users/user/.rbenv/versions/3.2.2/openssl/ssl/certs
Tools         
  Git         2.39.2 (Apple Git-143)
  RVM         not installed
  rbenv       rbenv 1.2.0
  chruby      0.3.9
```

## Bundler Build Metadata

```
Built At          2023-10-17
Git SHA           d10b46bd15
Released Version  true
```

## Bundler settings

```
force_ruby_platform
  Set for your local app (/Users/user/Documents/Awesome/.bundle/config): true
path
  Set for your local app (/Users/user/Documents/Awesome/.bundle/config): "vendor/bundle"
```

## Gemfile

### ../Gemfile

```ruby
source 'https://rubygems.org'

# You may use http://rbenv.org/ or https://rvm.io/ to install and use this version
ruby ">= 2.6.10"

gem 'cocoapods', '~> 1.13'
gem 'activesupport', '>= 6.1.7.3', '< 7.1.0'
```

### ../Gemfile.lock

```
GEM
  remote: https://rubygems.org/
  specs:
    CFPropertyList (3.0.6)
      rexml
    activesupport (7.0.8)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (>= 1.6, < 2)
      minitest (>= 5.1)
      tzinfo (~> 2.0)
    addressable (2.8.5)
      public_suffix (>= 2.0.2, < 6.0)
    algoliasearch (1.27.5)
      httpclient (~> 2.8, >= 2.8.3)
      json (>= 1.5.1)
    atomos (0.1.3)
    claide (1.1.0)
    cocoapods (1.13.0)
      addressable (~> 2.8)
      claide (>= 1.0.2, < 2.0)
      cocoapods-core (= 1.13.0)
      cocoapods-deintegrate (>= 1.0.3, < 2.0)
      cocoapods-downloader (>= 1.6.0, < 2.0)
      cocoapods-plugins (>= 1.0.0, < 2.0)
      cocoapods-search (>= 1.0.0, < 2.0)
      cocoapods-trunk (>= 1.6.0, < 2.0)
      cocoapods-try (>= 1.1.0, < 2.0)
      colored2 (~> 3.1)
      escape (~> 0.0.4)
      fourflusher (>= 2.3.0, < 3.0)
      gh_inspector (~> 1.0)
      molinillo (~> 0.8.0)
      nap (~> 1.0)
      ruby-macho (>= 2.3.0, < 3.0)
      xcodeproj (>= 1.23.0, < 2.0)
    cocoapods-core (1.13.0)
      activesupport (>= 5.0, < 8)
      addressable (~> 2.8)
      algoliasearch (~> 1.0)
      concurrent-ruby (~> 1.1)
      fuzzy_match (~> 2.0.4)
      nap (~> 1.0)
      netrc (~> 0.11)
      public_suffix (~> 4.0)
      typhoeus (~> 1.0)
    cocoapods-deintegrate (1.0.5)
    cocoapods-downloader (1.6.3)
    cocoapods-plugins (1.0.0)
      nap
    cocoapods-search (1.0.1)
    cocoapods-trunk (1.6.0)
      nap (>= 0.8, < 2.0)
      netrc (~> 0.11)
    cocoapods-try (1.2.0)
    colored2 (3.1.2)
    concurrent-ruby (1.2.2)
    escape (0.0.4)
    ethon (0.16.0)
      ffi (>= 1.15.0)
    ffi (1.16.3)
    fourflusher (2.3.1)
    fuzzy_match (2.0.4)
    gh_inspector (1.1.3)
    httpclient (2.8.3)
    i18n (1.14.1)
      concurrent-ruby (~> 1.0)
    json (2.6.3)
    minitest (5.20.0)
    molinillo (0.8.0)
    nanaimo (0.3.0)
    nap (1.1.0)
    netrc (0.11.0)
    public_suffix (4.0.7)
    rexml (3.2.6)
    ruby-macho (2.5.1)
    typhoeus (1.4.0)
      ethon (>= 0.9.0)
    tzinfo (2.0.6)
      concurrent-ruby (~> 1.0)
    xcodeproj (1.23.0)
      CFPropertyList (>= 2.3.3, < 4.0)
      atomos (~> 0.1.3)
      claide (>= 1.0.2, < 2.0)
      colored2 (~> 3.1)
      nanaimo (~> 0.3.0)
      rexml (~> 3.2.4)

PLATFORMS
  ruby

DEPENDENCIES
  activesupport (>= 6.1.7.3, < 7.1.0)
  cocoapods (~> 1.13)

RUBY VERSION
   ruby 3.2.2p53

BUNDLED WITH
   2.4.21
```

--- TEMPLATE END ----------------------------------------------------------------

Unfortunately, an unexpected error occurred, and Bundler cannot continue.

First, try this link to see if there are any existing issue reports for this error:
https://github.com/rubygems/rubygems/search?q=You+don%27t+have+write+permissions+for+the+%2FUsers%2Fuser%2FDocuments%2FAwesome%2Fvendor%2Fbundle%2Fruby%2F3.2.0+directory.&type=Issues

If there aren't any reports for this error yet, please fill in the new issue form located at https://github.com/rubygems/rubygems/issues/new?labels=Bundler&template=bundler-related-issue.md, and copy and paste the report template above in there.

**Comment by cipolleschi** [comment]
@Prabhat1933 which version of React Native are you using? And what command have you issued?

Please try:
- navigate to your project root 
- `rm -rf vendor/`
- `rm -rf Gemfile.lock`
- `cd ios`
- `bundle install`
- `bundle exec pod install`

Anyway, the error says `Gem::FilePermissionError: You don't have write permissions for the /Users/user/Documents/Awesome/vendor/bundle/ruby/3.2.0 directory` so it seems a different error, and a Permission one, so probably related to your local environment.

**Comment by Prabhat1933** [comment]
> @Prabhat1933 which version of React Native are you using? And what command have you issued?
> 
> Please try:
> 
> * navigate to your project root
> * `rm -rf vendor/`
> * `rm -rf Gemfile.lock`
> * `cd ios`
> * `bundle install`
> * `bundle exec pod install`
> 
> Anyway, the error says `Gem::FilePermissionError: You don't have write permissions for the /Users/user/Documents/Awesome/vendor/bundle/ruby/3.2.0 directory` so it seems a different error, and a Permission one, so probably related to your local environment.

Thank u so much it worked . I was Trying same step before from documentation but it was not working but this time it worked

**Comment by Prabhat1933** [comment]
nfo 💡 Tip: Make sure that you have set up your development environment correctly, by running react-native doctor. To read more about doctor command visit: https://github.com/react-native-community/cli/blob/main/packages/cli-doctor/README.md#doctor 

Exception in thread "main" javax.net.ssl.SSLHandshakeException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at java.base/sun.security.ssl.Alert.createSSLException(Alert.java:131)
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:360)
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:303)
        at java.base/sun.security.ssl.TransportContext.fatal(TransportContext.java:298)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1357)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(CertificateMessage.java:1232)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(CertificateMessage.java:1175)
        at java.base/sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392)
        at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:443)
        at java.base/sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:421)
        at java.base/sun.security.ssl.TransportContext.dispatch(TransportContext.java:183)
        at java.base/sun.security.ssl.SSLTransport.decode(SSLTransport.java:172)
        at java.base/sun.security.ssl.SSLSocketImpl.decode(SSLSocketImpl.java:1511)
        at java.base/sun.security.ssl.SSLSocketImpl.readHandshakeRecord(SSLSocketImpl.java:1421)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:456)
        at java.base/sun.security.ssl.SSLSocketImpl.startHandshake(SSLSocketImpl.java:427)
        at java.base/sun.net.www.protocol.https.HttpsClient.afterConnect(HttpsClient.java:580)
        at java.base/sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:201)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.followRedirect0(HttpURLConnection.java:2830)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.followRedirect(HttpURLConnection.java:2742)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1869)
        at java.base/sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1525)
        at java.base/sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:250)
        at org.gradle.wrapper.Download.downloadInternal(Download.java:109)
        at org.gradle.wrapper.Download.download(Download.java:89)
        at org.gradle.wrapper.Install$1.call(Install.java:83)
        at org.gradle.wrapper.Install$1.call(Install.java:63)
        at org.gradle.wrapper.ExclusiveFileAccessManager.access(ExclusiveFileAccessManager.java:69)
        at org.gradle.wrapper.Install.createDist(Install.java:63)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:109)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:66)
Caused by: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:439)
        at java.base/sun.security.validator.PKIXValidator.engineValidate(PKIXValidator.java:306)
        at java.base/sun.security.validator.Validator.validate(Validator.java:264)
        at java.base/sun.security.ssl.X509TrustManagerImpl.validate(X509TrustManagerImpl.java:313)
        at java.base/sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:222)
        at java.base/sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:129)
        at java.base/sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1341)
        ... 26 more
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
        at java.base/sun.security.provider.certpath.SunCertPathBuilder.build(SunCertPathBuilder.java:148)
        at java.base/sun.security.provider.certpath.SunCertPathBuilder.engineBuild(SunCertPathBuilder.java:129)
        at java.base/java.security.cert.CertPathBuilder.build(CertPathBuilder.java:297)
        at java.base/sun.security.validator.PKIXValidator.doBuild(PKIXValidator.java:434)
        ... 32 more
        
        can u help me with this error while building android emulator it is giving this error
        I have installed jdk zulu11 and set java Home environment according to the documentation but still it is giving this error and not able to build android app

**Comment by cipolleschi** [comment]
@Prabhat1933 this is a different issue, so, please, open a new one. Let's not pollute this issue for people looking at solution for Ruby failures.
